### PR TITLE
feat: add failure message manipulation functions

### PR DESCRIPTION
**Pull request by joseph-sentry**
This commit ports over two functions for manipulating failure messages:
- escape_failure_message:

  escapes a failure message to prepare it to be displayed in the comment so that the content of the failure message does not interfere with the formatting of the PR comment.

- shorten_file_paths:

  truncates the beginning of long file paths in the failure message in order to reduce the horizontal space a failure message takes up.

### CODE REVIEW

File: src/failure_message.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+lazy_static! {
+    static ref SHORTEN_PATH_PATTERN: Regex =
+        Regex::new(r"(?:\/*[\w\-]+\/)+(?:[\w\.]+)(?::\d+:\d+)*").unwrap();
+}
+
+#[pyfunction]
+pub fn shorten_file_paths(mut failure_message: String) -> String {
+    let original_failure_message = failure_message.clone();
```
Comment (giovanni-guidini) [inline src/failure_message.rs:36]: [nit] Wouldn't it make more sense to have the argument be immutable (which is generally a good idea) and have the local var be `mut` (and return that at the end)?

**Review by giovanni-guidini — COMMENTED** [review]
LGTM. There are some tests failing tho

**Review by matt-codecov — COMMENTED** [review]
in addition to your real-world-data test cases, i think some minimal, hand-written test cases would be useful too. the smaller inputs/expected outputs make good documentation comments + can make sure you cover specific behaviors that you do/don't want that don't necessarily show up in the real-world data

in this PR for example, i don't actually know whether your windows junit file has `\r` in it because some git/editor configurations normalize line endings automatically. it would be better to just test `\r` explicitly. also, what will this function do if the input has `\\r` or `\\n` in it? what would we want it to do? any other edge cases you can think of?

it's sort of "unit test" vs "integration test" thing

File: src/failure_message.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    failure_message: &PyString,
+) -> &'py PyString {
+    let mut escaped_failure_message = failure_message.to_string();
+    for (from, to) in REPLACEMENTS.entries() {
+        escaped_failure_message = escaped_failure_message.replace(from, to);
+    }
+    &PyString::new(py, &escaped_failure_message)
+}
+
+#[pyfunction]
```
Comment (giovanni-guidini) [inline src/failure_message.rs:33]: If I'm in python code is it possible to generate a rust string to pass to this function?

Also I didn't really understand why we need both of them exposed to the python code...

**Review by giovanni-guidini — APPROVED** [review]
